src/Checkers/ShapeTypeChecker.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Prezly\PropTypes\Checkers;
+
+use Prezly\PropTypes\Exceptions\PropTypeException;
+
+class ShapeTypeChecker implements TypeChecker
+{
+    /** @var \Prezly\PropTypes\Checkers\TypeChecker[] */
+    private $shape;
+
+    public function __construct(array $shape)
+    {
+        $this->shape = $shape;
```

Comment:
I suggest to add extra assertions around input argument. To make sure that the checks is guaranteed to have a valid state after it's constructed. This will simplify debugging errors to lib consumers.

```php
foreach ($shape_types as $key => $checker) {
    if (! $checker instanceof TypeChecker) {
        throw new InvalidArgumentException(sprintf(
            'Invalid argument supplied to shape(). Expected an associative array of %s instances, but received %s at key "%s".',
            TypeChecker::class,
            is_object($checker) ? get_class($checker) : gettype($checker),
            $key
        ));
    }
}
```